### PR TITLE
Performance & stability improvements

### PR DESCRIPTION
**Pull request by AlexanderSher**
- Improve DependencyResolver performance
- Fix memory leak in PythonAnalyzer
- Fix ImmutableArray performance

### CODE REVIEW

File: src/Analysis/Ast/Impl/Analyzer/PythonAnalyzer.cs [hunk 1]
```diff
@@ -123,6 +123,8 @@ public sealed class PythonAnalyzer : IPythonAnalyzer, IDisposable {
             lock (_syncObj) {
                 _analysisEntries.Remove(new AnalysisModuleKey(module));
             }
+
+            _dependencyResolver.Remove(new AnalysisModuleKey(module));
```
Comment (MikhailArkhipov) [inline src/Analysis/Ast/Impl/Analyzer/PythonAnalyzer.cs]: Looks we new it every time. Minor thing, but maybe just attach it to the module or have it produce it? Up to you.

**Comment by AlexanderSher** [comment]
Memory consumption for the project that imports `tensorflow`, `pandas` and `plotly`:
image

**Comment by jakebailey** [comment]
Using the example from #821, 2GB before, 968MB after. Nice.

**Comment by jakebailey** [comment]
I seem to be getting quite a few of these:

```
Analysis of sqlalchemy.engine.url(Library) failed. Exception message: Object reference not set to an instance of an object..
Analysis of PIL.GifImagePlugin(Library) completed in 75.6135 ms.
Analysis of sqlalchemy.sql.compiler(Library) failed. Exception message: Object reference not set to an instance of an object..
Analysis of numpy.ma.core(Library) failed. Exception message: Object reference not set to an instance of an object..
Analysis of numpy.ma.core(Library) failed. Exception message: Object reference not set to an instance of an object..
Analysis of scipy.optimize.nonlin(Library) failed. Exception message: Object reference not set to an instance of an object..
Analysis of scipy.integrate.quadrature(Library) failed. Exception message: Object reference not set to an instance of an object..
Analysis of scipy.optimize.nonlin(Library) failed. Exception message: Object reference not set to an instance of an object..
Analysis of PyQt5.QtWidgets(Compiled) completed in 15743.8101 ms.
Analysis of PIL.Image(Library) failed. Exception message: Object reference not set to an instance of an object..
Analysis of pandas.core.indexes.numeric(Library) failed. Exception message: Object reference not set to an instance of an object..
Analysis of pandas.core.indexes.numeric(Library) failed. Exception message: Object reference not set to an instance of an object..
Analysis version 4995 has been completed in 19650.2117 ms with 597 entries analyzed and 1773 entries skipped.
```

**Comment by AlexanderSher** [comment]
This is an unrelated issue: https://github.com/Microsoft/python-language-server/issues/1046

**Pull request merged**